lib/spack/spack/solver/asp.py

```diff
@@ -1536,18 +1538,14 @@ def versions_for(v):
     def define_compiler_version_constraints(self):
         compiler_list = spack.compilers.all_compiler_specs()
         compiler_list = list(sorted(set(compiler_list)))
+        constraints = set(c[1] for c in self.compiler_version_constraints)
 
-        for pkg_name, cspec in self.compiler_version_constraints:
+        for constraint in sorted(constraints):
```

Comment:
There's no reason to capture the pkg_name as part of the version constraints if we're not going to use it -- I think this could use the changes from #27964